File: app/config/cards_initcards.py
# -*- coding: utf-8 -*-

'''
Author: Hannibal
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
'''


cards_initcards_map = {};
cards_initcards_map[1] = {"lv":1,"count":5,"equip_group":"1,2","equip_min":1,"equip_max":4,"spell_group":"7.0",};


class Cards_initcards:
	def __init__(self, key):
		config = cards_initcards_map.get(key);
		for k, v in config.items():
			setattr(self, k, v);
		return

def create_Cards_initcards(key):
		config = cards_initcards_map.get(key);
		if not config:
			return
		return Cards_initcards(key)

File: tool/table/client/cards_initcards.ts
/*
Author: 
Data: 
Desc: local data config
NOTE: Don't modify this file, it's build by xml-to-python!!!
*/

module config{

let cards_initcards_map = null;
export function cards_initcards_map_init(config_obj:Object):void{
	cards_initcards_map = config_obj["cards_initcards_map"];
}


export class Cards_initcards extends Object{
	private m_config:any;
	public static m_static_map:Object = new Object();
	constructor(key){
		super();
		this.m_config = cards_initcards_map[key];
		for(let i in this.m_config){
			this[i] = this.m_config[i];
		}
		return;
	}
	public static get_Cards_initcards(key){
		if(Cards_initcards.m_static_map.hasOwnProperty(key) == false){
			Cards_initcards.m_static_map[key] = Cards_initcards.create_Cards_initcards(key);
		}
		return Cards_initcards.m_static_map[key];	
	}
	public static create_Cards_initcards(key){
		if(cards_initcards_map.hasOwnProperty(key)){
			return new Cards_initcards(key);
		}
		return null;	
	}
	public static get_cfg_object():Object
	{
		return cards_initcards_map;
	}
}
}